rfcs/convergence/native-element-props.md

```diff
@@ -185,36 +192,41 @@ There is no `root` prop:
 **Given JSX:**
 
 ```jsx
-<Checkbox id="myId" className="myClass" />
+<Checkbox id="myId" className="myClass" input={{ className: 'inputClass' }} />
 ```
 
 **Resulting DOM (simplified):**
 
 ```html
 <div class="myClass">
-  <input id="myId" />
+  <input id="myId" className="inputClass" />
 </div>
 ```
 
-If props are specified on the `root` or `input` slots, they always go to the specified slot, and always win over props specified on the element itself:
+To reduce confusion about which props take precedence, it's not allowed to specify props other than `className` and `style` on the primary slot:
+
+```jsx
+// ❌ Fails to compile
+<Checkbox input={{ id: 'inputId' }} />
+```
+
+However, explicitly specifying `className` and `style` on the `root` slot is allowed, and these will win over props specified on the element itself:
 
 **Given JSX:**
 
 ```jsx
 <Checkbox
-  id="myId" // ⚠ overridden by "inputId" below
   className="myClass" // ⚠ overridden by "rootClass" below
   style={{ color: 'red' }}
   root={{ id: 'rootId', className: 'rootClass' }}
-  input={{ id: 'inputId', className: 'inputClass' }}
```

Comment:
You should either leave this here without the `id` and with the `className`, or remove the `class` in the `input` tag in the resulting DOM below.